checks/__init__.py

```diff
@@ -531,8 +553,61 @@ def get_warnings(self):
         self.warnings = []
         return warnings
 
+    @staticmethod
+    def _get_statistic_name_from_method(method_name):
+        return method_name[4:] if method_name.startswith('get_') else method_name
+
+    @staticmethod
+    def _collect_stats(methods=None, log=None):
+        current_process = psutil.Process(os.getpid())
+
+        methods = methods or DEFAULT_PSUTIL_METHODS
+        filtered_methods = [m for m in methods if hasattr(current_process, m)]
+
+        stats = {}
+
+        for method in filtered_methods:
+            # Go from `get_memory_info` -> `memory_info`
+            stat_name = AgentCheck._get_statistic_name_from_method(method)
+            try:
+                raw_stats = getattr(current_process, method)()
+                try:
+                    stats[stat_name] = raw_stats._asdict()
+                except AttributeError:
+                    if isinstance(raw_stats, numbers.Number):
+                        stats[stat_name] = raw_stats
+                    else:
+                        if log is not None:
+                            log.warn("Could not serialize output of {0} to dict".format(method))
+
+            except psutil.AccessDenied:
+                if log is not None:
+                    log.warn("Cannot call psutil method {} : Access Denied".format(method))
+
+        return stats
+
+    def set_stats(self, before, after):
+        self.stats = {'before': before, 'after': after}
+
+    def get_stats(self):
```

Comment:
Watchout:
```python
get_stats
```
name 
is likely to be used inside checks.d.
It's already used by https://github.com/DataDog/dd-agent/blob/master/checks.d/rabbitmq.py#L162
And is likely used by some custom checks.d. 

You should use something different such as:
```python
def _get_internal_profiling_stats(...):
 ```